CountryPicker/CountryPicker/PickerView/CountryPickerView.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  CountryPickerView.swift
+//  CountryPicker
+//
+//  Created by Hardeep Singh on 05/12/19.
+//  Copyright © 2019 SuryaKant Sharma. All rights reserved.
+//
+
+import UIKit
+
+//public protocol ActionView {
+//    func showOnView(countryPicker: CountryPickerView)
+//}
+
+open class CountryPickerView: UIPickerView {
+    
+    private var selectedCountry: Country? {
+        didSet {
+            self.scrollToSelectedCountry()
+        }
+    }
+    
+    // ISO 3166-1 alpha-2 two-letter country codes.
+    private var countryCode: [String] = [String]() {
+        didSet {
+            self.updatePickList()
+        }
+    }
+    
+    private var allCountryList: [Country] = [Country]()
+    private var pickList: [Country] = [Country]() {
+        didSet {
+            self.reloadComponent(0)
+        }
+    }
+    
+    private func updatePickList() {
+        guard countryCode.count > 0 else {
+            return
+        }
+        
+        var tempList = countryCode
+        
+        let list = allCountryList.filter { (country) -> Bool in
+            if let index = tempList.firstIndex(of: country.countryCode) {
+                tempList.remove(at: index)
+                return true
+            }
+            return false
+        }
+        
+        self.pickList = list
+        
+    }
+
+    init() {
+        super.init(frame: CGRect.zero)
+        self.configure()
+    }
+    
+    required public init?(coder: NSCoder) {
+        super.init(coder: coder)
+        configure()
+    }
+    
+    private var didSelectCountryCallback: ((_ country: Country) -> Void)?
+    public func onSelectCountry(callback: @escaping (_ country: Country) -> Void) {
+        self.didSelectCountryCallback = callback
+    }
+    
+    private func configure() {
+        
+        self.selectedCountry = CountryManager.shared.currentCountry
+        self.allCountryList = CountryManager.shared.countries
+        self.pickList = allCountryList
+        
+        self.delegate = self
+        self.dataSource = self
+        
+    }
+     
+    deinit {
+        print("Picker view nil...")
+    }
```

Comment:
Please remove deinit and print statement if you are sure that no memory leaks are there